alf/algorithms/off_policy_algorithm.py

```diff
@@ -150,9 +150,13 @@ def _train_experience(self, experience, num_updates, mini_batch_size,
 
         length = experience.step_type.shape[1]
         mini_batch_length = (mini_batch_length or length)
-        assert length % mini_batch_length == 0, (
-            "length=%s not a multiple of mini_batch_length=%s" %
-            (length, mini_batch_length))
+        if length % mini_batch_length:
+            common.warning_once(
+                "length=%s not a multiple of mini_batch_length=%s" %
+                (length, mini_batch_length))
+            length = length // mini_batch_length * mini_batch_length
+            common.warning_once(
+                "Experience length has been cut to %s" % length)
```

Comment:
Where is the logic for cutting experience?

Reply:
length = length // mini_batch_length * mini_batch_length